src/core_codemods/harden_pyyaml.py

```diff
@@ -50,18 +53,70 @@ def rule(cls):
                                 - pattern: yaml.BaseLoader
                                 - pattern: yaml.FullLoader
                                 - pattern: yaml.UnsafeLoader
+                  - patterns:
+                      - pattern: |
+                          class $X(...,$LOADER, ...):
+                            ...
+                      - metavariable-pattern:
+                          metavariable: $LOADER
+                          patterns:
+                            - pattern-either:
+                                - pattern: yaml.Loader
+                                - pattern: yaml.BaseLoader
+                                - pattern: yaml.FullLoader
+                                - pattern: yaml.UnsafeLoader
 
         """
 
-    def on_result_found(self, original_node, updated_node):
-        maybe_name = self.get_aliased_prefix_name(original_node, self._module_name)
-        maybe_name = maybe_name or self._module_name
-        if maybe_name == self._module_name:
-            self.add_needed_import(self._module_name)
-        new_args = [
-            *updated_node.args[:1],
-            updated_node.args[1].with_changes(
-                value=self.parse_expression(f"{maybe_name}.SafeLoader")
-            ),
-        ]
-        return self.update_arg_target(updated_node, new_args)
+    def on_result_found(
+        self,
+        original_node: Union[cst.Call, cst.ClassDef],
+        updated_node: Union[cst.Call, cst.ClassDef],
+    ):
+        match original_node:
+            case cst.Call():
```

Comment:
It would actually be useful to provide different change descriptions for each of the cases here but I'm not quite sure that `on_result_found` as currently written enables us to do that. At the very least can you add a TODO here?